### ISSUE TITLE
<Icon /> does not forward the ref

### ISSUE THREAD

**Issue by hrasoa** [iconify/iconify#57]
I should get the svg dom element in the ref.

Demo:

https://9ghol.csb.app/

```
index.js:27 Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Check the render method of `App`.
    at Icon$1 (https://9ghol.csb.app/node_modules/@iconify/react-with-api/dist/icon.esm.js:2054:10)
    at div
    at App (https://9ghol.csb.app/src/App.js:24:34)
```

**Comment by cyberalien** [comment]
Good catch! Thanks for reporting it.

I'll try to fix that as soon as possible.

**Comment by cyberalien** [comment]
Fixed in `@iconify/react@2.0.0-rc.5`.

Couldn't figure out how to unit test it, so it comes without unit tests. However, in `react-demo` package I've added visual test to make sure it works correctly.

Unfortunately couldn't figure out how to pass it from class to child component, so `@iconify/react-with-api` doesn't support references yet. I'll try to figure it out.

**Comment by hrasoa** [comment]
Great, thanks :+1:

**Comment by cyberalien** [comment]
I'm really struggling with implementing this with API package. 

Problem is, component could be rendered instantly or it could render asynchronously, which means `render()` might return `createElement('span', props)` or it could return `null`. Additionally I'm very ill right now, my brain barely functions, which makes it very hard to come up with solutions.

If you have an idea on how to implement it, please help. Without going into Iconify internal functionality, it just needs to be a component could render synchronously and asynchronously (depending if data has loaded) and passes reference to generated svg (content can be anything, I can adapt code for svg). Currently component is using classes, but using hooks will work fine too (I'm using class because I'm not familiar with hooks enough to create stable component).

**Comment by cyberalien** [comment]
Fixed in 3.0.0-dev (currently available in `react-restructure` branch of this repo). Now API package also supports references, which required full rewrite of component.

**Comment by hrasoa** [comment]
Awesome, thank you !
